src/components/vm/snapshots/vmSnapshotsCreateModal.jsx

```diff
@@ -64,41 +69,80 @@ const DescriptionRow = ({ onValueChanged, description }) => {
     );
 };
 
-function getDefaultMemoryPath(vm, snapName) {
-    // Choosing a default path where memory snapshot should be stored might be tricky. Ideally we want
-    // to store it in the same directory where the primary disk (the disk which is first booted) is stored
-    // If however no such disk can be found, we should fallback to libvirt's default /var/lib/libvirt
+function getDefaultMemoryLocation(vm) {
+    // If we find an existing external snapshot, use it's memory path
+    // as the default. Otherwise, try to find the primary disk and use
+    // it's location. If that fails as well, use a reasonable hard
+    // coded value.
+
+    for (const s of vm.snapshots.sort((a, b) => b.creationTime - a.creationTime)) {
+        if (s.memoryPath)
+            return dirname(s.memoryPath);
+    }
+
     const devices = getSortedBootOrderDevices(vm).filter(d => d.bootOrder &&
                                                               d.device.device === "disk" &&
                                                               d.device.type === "file" &&
                                                               d.device.source.file);
     if (devices.length > 0) {
         const primaryDiskPath = devices[0].device.source.file;
-        const directory = primaryDiskPath.substring(0, primaryDiskPath.lastIndexOf("/") + 1);
-        return directory + snapName;
+        const directory = dirname(primaryDiskPath);
+        return directory;
     } else {
         if (vm.connectionName === LIBVIRT_SYSTEM_CONNECTION)
-            return "/var/lib/libvirt/memory/" + snapName;
+            return "/var/lib/libvirt/memory";
         else if (current_user)
-            return current_user.home + "/.local/share/libvirt/memory/" + snapName;
+            return current_user.home + "/.local/share/libvirt/memory";
     }
 
     return "";
 }
 
-const MemoryPathRow = ({ onValueChanged, memoryPath, validationError }) => {
+const MemoryLocationRow = ({ onValueChanged, memoryLocation, validationError, available, needed }) => {
+    let info = "";
+    let info_variant = "default";
+
+    if (available) {
+        info = cockpit.format(_("$0 available at this location."), formatWithBestUnit(available));
+        if (available * 0.9 < needed)
+            info_variant = "warning";
```

Comment:
Below you check `if (needed)` so it's optional. `Number < undefined` is always false, so without `needed` this will always be variant `default`. This seems correct to me, just a bit unclean to rely on such quirky semantics.

Reply:
Yes, good point.